content/snippets/capacity-planning.md

```diff
@@ -0,0 +1,95 @@
+
+#### SSD Storage
+
+SSD Storage should support several thousand both read and write IOPS. Maximum read IOPS come from heavily used servers such as those behind https://s1.ripple.com and https://s2.ripple.com, which service transaction and ledger history retrieval requests. This is sometimes over 10,000 reads per second! Maximum write IOPS have come from performance testing in which up to 7,000 writes/s have been observed.
+
+#### CPU Utilization and Virtualization
+Ripple perfmance engineering finds that bare metal servers acheive maximum throughput. However, it is likely that hypervisors would cause minimal degradation in performance.
+
+#### Network
+
+Any enterprise or carrier-class data center should have plenty of network bandwidth to support running `rippled` servers. The minimum requirements are roughly 2Mbps transmit and 2Mbps receive for current transaction volumes. However, these can burst up to 100MBps transmissions when serving up historical ledger and transaction reports. When a `rippled` server initially starts up, it can burst to well over 20Mbps receive.
+
+#### Storage
+
+Ripple recommends estimating storage sizing at roughly 12GB per day of data kept online with NuDB. RockDB requires around 8GB per day. However, future storage requirements are not known and extra capacity should be available to account for future growth. Currently, a server with all XRP Ledger history requires 6.8TB.
+
+<!-- {# ***TODO: Update the dated storage consideration above, as needed. ***#} -->
+<!-- {# ***TODO: Create historic metrics that a user can use to derive what will be required. For ex, a chart with 1TB in 2014, 3TB in 2015, 7TB in 2018 ***#} -->
+
+#### Memory
+
+Memory requirements are mainly a function of the `node_size` configuration option and the amount of client traffic retrieving historical data. As mentioned, production servers should maximize performance and set this parameter to `huge`. A smaller memory footprint can be achieved by setting the parameter lower, but that should only be used for testing. With a `node_size` of `medium`, a `rippled` server can be reasonably stable in a test Linux system with as little as 8GB of RAM.
+
+#### Amazon Web Servives
+
+Amazon Web Services (AWS) is a popular virtualized hosting environment, and you can run `rippled`  in that environment. However, using EBS storage is not recommended, because the maximum number of IOPS (5,000) may not be able to keep up with the heaviest demands, yet it is very expensive.
+
+AWS instance stores (`ephemeral` storage) do not have these constraints. Therefore, Ripple recommends deploying `rippled` servers with host types such as `M3` that have instance storage. The `database_path` and `node_db` path should each reside on instance storage.
+
+**Caution:** AWS instance storage is not guaranteed to provide durability in the event of hard drive failure. Further, data will be lost when the instance is stopped and restarted (but not when just rebooted).
```

Comment:
I suspect this may not _actually_ be a "caution" situation. You should add a comment explaining why this ephemeral storage is OK for `rippled`'s databases:

> AWS instance storage is not guaranteed in the event of hard drive failure. Further, data is lost whenever the instance is stopped and restarted (although not if it is directly rebooted). However, this is usually acceptable for a `rippled` server because any individual server doesn't store any critical, unique data. At most, a `rippled` server may need to generate a new unique server ID when it reconnects to the network, re-download the configured amount of ledger history from its peer servers, and it may lose track of some unconfirmed transactions that were queued at the time it shut down.

Reply:
For reference, when I say "unique server ID" I'm referring to the cryptographic keys that every server uses to identify itself and sign its communications in peer-to-peer communications. (The public key of this keypair is the `pubkey_node` from server_info.) The server generates this key when it starts up and re-generates it if it doesn't have one. It's generally not important whether a server changes IDs this way. The only exception would be if you have your servers configured for [clustering](https://ripple.com/build/rippled-setup/#clustering) in which case the other members of the cluster won't recognize the server by its new ID. You can recognize this by hard-coding the server's ID in the config file using the `[node_seed]` stanza.

Reply:
Great, that explanation (your second comment) makes sense, but I don't think we need to get into server ID's here, rather just manage the data loss exposure. I'll take a middle approach to that with my update.

Reply:
> **Caution:** AWS instance storage is not guaranteed to provide durability in the event of hard drive failure. Further, data will be lost when the instance is stopped and restarted (but not when just rebooted). This loss can be acceptable for a `rippled` server because an individual server can usually re-acquire that data from its peer servers.